connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/ClusterConfigState.java

```diff
@@ -112,12 +112,13 @@ public boolean contains(String connector) {
      * {@link org.apache.kafka.common.config.ConfigTransformer} by having all variable
      * references replaced with the current values from external instances of
      * {@link ConfigProvider}, and may include secrets.
+     * NOTED: If the configuration is not null, it will have been transformed.
      * @param connector name of the connector
      * @return a map containing configuration parameters
      */
     public Map<String, String> connectorConfig(String connector) {
         Map<String, String> configs = connectorConfigs.get(connector);
-        if (configTransformer != null) {
+        if (configs != null && configTransformer != null) {
```

Comment:
I wonder if a better location for `null` checking configs is inside `transform`. To me, it seems it is, because this way the check is in one place and we are protected against future uses of `tranform` with a `null` argument.
 
Also, in contrast, seems that `configTransformer` can not be `null`.

Reply:
bq. Also, in contrast, seems that configTransformer can not be null.
Not really. In DistributedHerder's constructor, the first configState is ClusterConfigState.EMPTY. And the ClusterConfigState.EMPTY has a null configTransformer.

We can introduce a interface of WorkerConfigTransformer and then pass a do-nothing impl to create an empty ClusterConfigState. Consequently, we can make the configTransformer in ClusterConfigState non-nullable.

Reply:
> I wonder if a better location for null checking configs is inside transform. To me, it seems it is, because this way the check is in one place and we are protected against future uses of tranform with a null argument.

It makes sense to me. Will address this comment in next commit